pkg/events/parse/params_test.go

```diff
@@ -0,0 +1,402 @@
+package parse
+
+import (
+	"testing"
+
+	"github.com/aquasecurity/tracee/types/trace"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestArgInt32Val(t *testing.T) {
+
+	tests := []struct {
+		name          string
+		arg           trace.Argument
+		expectedValue int32
+		errorMessage  string
+	}{
+		{
+			name: "valid_val",
+			arg: trace.Argument{
+				ArgMeta: trace.ArgMeta{
+					Name: "valid_val",
+					Type: "int",
+				},
+				Value: int32(1878),
+			},
+			expectedValue: int32(1878),
+		},
+		{
+			name: "invalid_val",
+			arg: trace.Argument{
+				ArgMeta: trace.ArgMeta{
+					Name: "invalid_val",
+					Type: "int",
+				},
+				Value: int64(1878),
+			},
+			errorMessage: "argument invalid_val is not of type int32",
+		},
+		{
+			name: "no_val",
+			arg: trace.Argument{
+				ArgMeta: trace.ArgMeta{
+					Name: "does_not_exist_val",
+					Type: "int",
+				},
+				Value: int32(1878),
+			},
+			errorMessage: "argument no_val not found",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := trace.Event{Args: []trace.Argument{
+				{
+					ArgMeta: trace.ArgMeta{
+						Name: tt.arg.Name,
+						Type: tt.arg.Type,
+					},
+					Value: tt.arg.Value,
+				},
+			}}
+			val, err := ArgInt32Val(&e, tt.name)
+			if tt.errorMessage != "" {
+				assert.Equal(t, tt.errorMessage, err.Error())
+				assert.Error(t, err)
+			} else {
+				assert.Equal(t, tt.expectedValue, val)
+				assert.NoError(t, err)
```

Comment:
check no error before checking expected value
```suggestion
				assert.NoError(t, err)
				assert.Equal(t, tt.expectedValue, val)
```

Reply:
Done!